internal/tfconfig/remote_state.go

```diff
@@ -0,0 +1,34 @@
+package tfconfig
+
+// AddRemoteStates adds remote state data sources to the module
+// TODO: discard this entirely in favor of AppendResource
+// also: the logic here is incorrect, this func is setting, it is not adding/appending
```

Comment:
seems like the name is more wrong than the logic. Would something like SetRemoteStates have been a better choice?

Reply:
I think `Add*` is a better API. It's unlikely someone really wants to set and blow away existing changes. The best API is probably `AddResource` and `AddDataSource` in `tfconfig` and just looping over remote state objects inline and calling those lower level funcs.